PDFView fails to load local PDFs

**Describe the bug**
An exception is thrown when local pdfs get rendered.

The error geths thrown in this line: https://github.com/rumax/react-native-PDFView/blob/91471635edba98b87c7a462465c34a7f71758f4e/android/src/main/java/com/rumax/reactnative/pdfviewer/AsyncDownload.java#L55

```
11-27 13:07:47.613  4479  4520 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at android.os.AsyncTask$3.done(AsyncTask.java:353)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:764)
11-27 13:07:47.613  4479  4520 E AndroidRuntime: Caused by: java.lang.ClassCastException: sun.net.www.protocol.file.FileURLConnection cannot be cast to java.net.HttpURLConnection
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at com.rumax.reactnative.pdfviewer.AsyncDownload.doInBackground(AsyncDownload.java:55)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at com.rumax.reactnative.pdfviewer.AsyncDownload.doInBackground(AsyncDownload.java:24)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at android.os.AsyncTask$2.call(AsyncTask.java:333)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-27 13:07:47.613  4479  4520 E AndroidRuntime:        ... 4 more

```

I see no problem with making this compatible with other url types. Maybe just handle the HTTPConnection case differently and revert the other way.

**To Reproduce**
Load a PDF from a local file url

**Expected behavior**
File should be loaded

**Smartphone:**
 - OS: Android

**Additional context**

This feature was implemented some time ago: https://github.com/rumax/react-native-PDFView/issues/42

--- a/ReactNativeViewPDF/PDFConstants.h
+++ b/ReactNativeViewPDF/PDFConstants.h
@@ -13,6 +13,10 @@ extern NSString * const URL_PROPS_METHOD_KEY;
 extern NSString * const URL_PROPS_HEADERS_KEY;
 extern NSString * const URL_PROPS_BODY_KEY;
 
+// Supported protocols
+extern NSString * const HTTP;
+extern NSString * const HTTPS;
+
 // Error types
 extern NSString * const ERROR_UNSUPPORTED_TYPE;
 extern NSString * const ERROR_ONLOADING;
@@ -21,5 +25,6 @@ extern NSString * const ERROR_INVALID_REACT_TAG;
 
 // Error messages
 extern NSString * const ERROR_MESSAGE_KEY;
+extern NSString * const ERROR_MESSAGE_TITLE;
 extern NSString * const ERROR_MESSAGE_BASE64_NIL;
 extern NSString * const ERROR_MESSAGE_FILENOTFOUND;

--- a/ReactNativeViewPDF/PDFConstants.m
+++ b/ReactNativeViewPDF/PDFConstants.m
@@ -13,6 +13,10 @@ NSString *const URL_PROPS_METHOD_KEY = @"method";
 NSString *const URL_PROPS_HEADERS_KEY = @"headers";
 NSString *const URL_PROPS_BODY_KEY = @"body";
 
+// Supported protocols
+NSString *const HTTP = @"http";
+NSString *const HTTPS = @"https";
+
 // Error types
 NSString *const ERROR_UNSUPPORTED_TYPE = @"Unsupported resourceType";
 NSString *const ERROR_ONLOADING = @"Error occured while loading content in webview";
@@ -20,6 +24,7 @@ NSString *const ERROR_REQUIRED_INPUT_NOT_SET = @"Validation failed. Confirm reso
 NSString *const ERROR_INVALID_REACT_TAG = @"Invalid react tag";
 
 // Error messages
-NSString *const ERROR_MESSAGE_KEY = @"errorMessage";
+NSString *const ERROR_MESSAGE_KEY = @"message";
+NSString *const ERROR_MESSAGE_TITLE = @"title";
 NSString *const ERROR_MESSAGE_BASE64_NIL = @"Converted Base64 data is nil";
 NSString *const ERROR_MESSAGE_FILENOTFOUND = @"PDF file not found";

--- a/ReactNativeViewPDF/PDFView.m
+++ b/ReactNativeViewPDF/PDFView.m
@@ -18,7 +18,7 @@
     self = [super init];
     if ( self ) {
         [self setBackgroundColor: [UIColor clearColor]];
-        
+
         webview = [[WKWebView alloc] initWithFrame: self.frame];
         [webview setNavigationDelegate: self];
         [self addSubview: webview];
@@ -32,31 +32,31 @@
 
 - (void)layoutSubviews {
     [super layoutSubviews];
-    
+
     webview.frame = self.bounds;
     didLoadOnce = true;
-    
+
     if (![self isNewInput]) {
         return;
     }
-    
+
     [self renderContent];
 }
 
 - (void)renderContent {
     [webview setAlpha: 0.0];
-    
+
     if (![self isRequiredInputSet]) {
         return;
     }
-    
+
     [self updateInput];
-    
+
     if (![self isSupportedResourceType]) {
         [self throwError: ERROR_UNSUPPORTED_TYPE withMessage: [NSString stringWithFormat: @"resourceType: %@ not recognized", _resourceType]];
         return;
     }
-    
+
     if ([self isURLResource]) {
         [webview loadRequest: [self createRequest]];
     } else if ([self isFileResource]) {
@@ -68,7 +68,7 @@
             [self throwError: ERROR_ONLOADING withMessage: ERROR_MESSAGE_FILENOTFOUND];
             return;
         }
-        
+
         [webview loadFileURL: targetURL allowingReadAccessToURL: targetURL];
     } else {
         NSString *characterEncodingName = [_textEncoding isEqual: UTF_16] ? UTF_16 : UTF_8;
@@ -82,13 +82,22 @@
 }
 
 - (NSMutableURLRequest *)createRequest {
-    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL: [NSURL URLWithString: _resource]
+    NSURL *URL = [NSURL URLWithString: _resource];
+    NSString *scheme = URL.scheme;
+
+    if ([HTTP caseInsensitiveCompare:scheme] != NSOrderedSame &&
+        [HTTPS caseInsensitiveCompare:scheme] != NSOrderedSame) {
+        NSString *errorMessage = [NSString stringWithFormat:@"Protocol %@ is not supported", scheme];
+        [self throwError: errorMessage withMessage: errorMessage];
+    }
+
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL: URL
                                                            cachePolicy: NSURLRequestUseProtocolCachePolicy
                                                        timeoutInterval: 60.0];
     if (_urlProps != nil) {
         [self enrichRequestWithUrlProps: request];
     }
-    
+
     return request;
 }
 
@@ -96,7 +105,7 @@
     NSString* method = [_urlProps objectForKey: URL_PROPS_METHOD_KEY];
     NSString* body = [_urlProps objectForKey: URL_PROPS_BODY_KEY];
     NSDictionary* headers = [_urlProps objectForKey: URL_PROPS_HEADERS_KEY];
-    
+
     if (method != nil) {
         [request setHTTPMethod: method];
     }
@@ -119,7 +128,7 @@
     [UIView animateWithDuration: _fadeInDuration animations: ^(void) {
         [webview setAlpha: 1.0];
     }];
-    
+
     if (_onLoad) {
         _onLoad(nil);
     }
@@ -183,9 +192,8 @@
 - (void)throwError: (NSString *)title withMessage:(NSString *)message {
     if (_onError) {
         _onError(@{
-                   title: @{
-                           ERROR_MESSAGE_KEY: message,
-                           }
+                   ERROR_MESSAGE_KEY: message,
+                   ERROR_MESSAGE_TITLE: title,
                    });
     }
 }

--- a/android/src/main/java/com/rumax/reactnative/pdfviewer/AsyncDownload.java
+++ b/android/src/main/java/com/rumax/reactnative/pdfviewer/AsyncDownload.java
@@ -22,6 +22,8 @@ import java.net.HttpURLConnection;
 import java.net.URL;
 
 class AsyncDownload extends AsyncTask<Void, Void, Void> {
+    public static final String HTTP = "http";
+    public static final String HTTPS = "https";
     private static final int BUFF_SIZE = 8192;
     private static final String PROP_METHOD = "method";
     private static final String PROP_BODY = "body";
@@ -30,7 +32,7 @@ class AsyncDownload extends AsyncTask<Void, Void, Void> {
     private TaskCompleted listener;
     private File file;
     private String url;
-    private IOException exception;
+    private Exception exception;
 
     AsyncDownload(String url, File file, ReadableMap urlProps, TaskCompleted listener) {
         this.listener = listener;
@@ -52,10 +54,15 @@ class AsyncDownload extends AsyncTask<Void, Void, Void> {
 
         try {
             url = new URL(this.url);
+            String protocol = url.getProtocol();
+            if (!protocol.equalsIgnoreCase(HTTP) && !protocol.equalsIgnoreCase(HTTPS)) {
+                exception = new IOException("Protocol \"" + protocol + "\" is not supported");
+                return null;
+            }
             connection = (HttpURLConnection) url.openConnection();
             enrichWithUrlProps(connection);
             connection.connect();
-        } catch (IOException e) {
+        } catch (Exception e) {
             exception = e;
             return null;
         }
@@ -81,11 +88,7 @@ class AsyncDownload extends AsyncTask<Void, Void, Void> {
 
     @Override
     protected void onPostExecute(Void param) {
-        if (exception == null) {
-            listener.downloadTaskCompleted();
-        } else {
-            listener.downloadTaskFailed(exception);
-        }
+        listener.onComplete(exception);
     }
 
     private void enrichWithUrlProps(HttpURLConnection connection) throws IOException {
@@ -156,7 +159,6 @@ class AsyncDownload extends AsyncTask<Void, Void, Void> {
     }
 
     public interface TaskCompleted {
-        void downloadTaskCompleted();
-        void downloadTaskFailed(IOException e);
+        void onComplete(Exception ex);
     }
 }

--- a/android/src/main/java/com/rumax/reactnative/pdfviewer/PDFView.java
+++ b/android/src/main/java/com/rumax/reactnative/pdfviewer/PDFView.java
@@ -27,9 +27,9 @@ import java.io.IOException;
 import java.io.InputStream;
 
 public class PDFView extends com.github.barteksc.pdfviewer.PDFView implements
-        OnLoadCompleteListener,
         OnErrorListener,
-        OnPageChangeListener {
+        OnPageChangeListener,
+        OnLoadCompleteListener {
     public final static String EVENT_ON_LOAD = "onLoad";
     public final static String EVENT_ON_ERROR = "onError";
     public final static String EVENT_ON_PAGE_CHANGED = "onPageChanged";
@@ -143,14 +143,13 @@ public class PDFView extends com.github.barteksc.pdfviewer.PDFView implements
 
         downloadTask = new AsyncDownload(resource, downloadedFile, urlProps, new AsyncDownload.TaskCompleted() {
             @Override
-            public void downloadTaskCompleted() {
-                renderFromFile(downloadedFile.getAbsolutePath());
-            }
-
-            @Override
-            public void downloadTaskFailed(IOException e) {
-                cleanDownloadedFile();
-                onError(e);
+            public void onComplete(Exception ex) {
+                if (ex == null) {
+                    renderFromFile(downloadedFile.getAbsolutePath());
+                } else {
+                    cleanDownloadedFile();
+                    onError(ex);
+                }
             }
         });
         downloadTask.execute();

--- a/demo/package.json
+++ b/demo/package.json
@@ -10,7 +10,7 @@
     "react-native": "0.57.4",
     "react-native-dropdownalert": "^3.7.0",
     "react-native-loading-spinner-overlay": "^1.0.1",
-    "react-native-view-pdf": "0.6.0"
+    "react-native-view-pdf": "0.6.1"
   },
   "devDependencies": {
     "babel-eslint": "^9.0.0",

--- a/demo/src/App.js
+++ b/demo/src/App.js
@@ -16,7 +16,7 @@ import Button from './Button';
 import pkg from '../package.json';
 
 type StateType = {
-  activeButton?: 'assets' | 'file' | 'url' | 'post' | 'base64' | 'error',
+  activeButton?: 'assets' | 'file' | 'url' | 'post' | 'base64' | 'errorData' | 'errorProtocol',
   resource?: Resource,
   spinner: boolean,
   canReload?: boolean,
@@ -114,8 +114,17 @@ export default class App extends React.Component<*, StateType> {
 
   dataWithError = () => {
     this.setState({
-      activeButton: 'error',
-      resource: resources.invalid,
+      activeButton: 'errorData',
+      resource: resources.invalidData,
+      spinner: true,
+    });
+  }
+
+
+  protocolWithError = () => {
+    this.setState({
+      activeButton: 'errorProtocol',
+      resource: resources.invalidProtocol,
       spinner: true,
     });
   }
@@ -198,8 +207,8 @@ export default class App extends React.Component<*, StateType> {
           <Button active={activeButton === 'url'} onPress={this.setUrl} title="Url" />
           <Button active={activeButton === 'base64'} onPress={this.setBase64} title="Base64" />
           <Button active={activeButton === 'file'} onPress={this.setFile} title="File" />
-          <Button active={activeButton === 'error'} onPress={this.dataWithError} title="Error" />
-          <Button active={activeButton === 'reset'} onPress={this.resetData} title="Reset" />
+          <Button active={activeButton === 'assets'} onPress={this.setFileAssets} title="Assets" />
+          <Button active={activeButton === 'post'} onPress={this.setUrlPost} title="Url Post" />
         </View>
         <PdfContent
           resource={state.resource}
@@ -209,8 +218,8 @@ export default class App extends React.Component<*, StateType> {
           onPageChanged={this.handlePageChanged}
         />
         <View style={styles.tabs}>
-          <Button active={activeButton === 'post'} onPress={this.setUrlPost} title="Url Post" />
-          <Button active={activeButton === 'assets'} onPress={this.setFileAssets} title="Assets" />
+          <Button active={activeButton === 'errorData'} onPress={this.dataWithError} title="Error data" />
+          <Button active={activeButton === 'errorProtocol'} onPress={this.protocolWithError} title="Error protocol" />
           <Button active={activeButton === 'reset'} onPress={this.resetData} title="Reset" />
         </View>
         {state.canReload && (

--- a/demo/src/resources.js
+++ b/demo/src/resources.js
@@ -41,9 +41,13 @@ const resources: {[key: string]: Resource } = {
     resource: base64Data.document,
     resourceType: 'base64',
   },
-  invalid: {
+  invalidData: {
     resource: '**invalid base 64**',
     resourceType: 'base64',
+  },
+  invalidProtocol: {
+    resource: 'file:/test-pdf.pdf',
+    resourceType: 'url',
   },
 };
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-native-view-pdf",
-  "version": "0.6.0",
+  "version": "0.6.1",
   "description": "React native Pdf viewer implementation",
   "main": "src/index.js",
   "scripts": {

